File: leetcode/twopointer/sortColors.ts
function sortColors(nums: number[]): void {
  const numCount = {
    0: 0,
    1: 0,
    2: 0,
  };
  for (let i = 0; i < nums.length; i++) {
    numCount[nums[i]]++;
  }
  for (let i = 0; i < nums.length; i++) {
    if (numCount[0] > 0) {
      nums[i] = 0;
      numCount[0]--;
    } else if (numCount[1] > 0) {
      nums[i] = 1;
      numCount[1]--;
    } else {
      nums[i] = 2;
    }
  }
}

File: leetcode/array/sortColors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        numsMap = {
            0: 0,
            1: 0,
            2: 0,
        }
        for n in nums:
            if n in numsMap:
                numsMap[n] += 1
        
        idx = 0
        for key, value in numsMap.items():
            for i in range(value):
                nums[idx]= key
                idx += 1
        